File: CS/ValidateRow_MVVM/MainViewModel.cs
using DevExpress.Mvvm;
using DevExpress.Mvvm.DataAnnotations;
using DevExpress.Mvvm.Xpf;
using System;
using System.Collections.Generic;
using System.ComponentModel;

namespace ValidateRow_MVVM {
    public class Task : IDataErrorInfo {
        public string TaskName { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }

        string IDataErrorInfo.Error => StartDate > EndDate ? "Either Start Date or End Date should be corrected." : null;
        string IDataErrorInfo.this[string columnName] {
            get {
                switch(columnName) {
                    case nameof(StartDate):
                        if(StartDate > EndDate)
                            return "Start Date must be less than End Date";
                        break;
                    case nameof(EndDate):
                        if(StartDate > EndDate)
                            return "End Date must be greater than Start Date";
                        break;
                    case nameof(TaskName):
                        if(string.IsNullOrEmpty(TaskName))
                            return "Enter a task name";
                        break;
                }
                return null;
            }
        }
    }

    public class MainViewModel : ViewModelBase {
        public List<Task> TaskList { get; }
        public MainViewModel() {
            TaskList = new List<Task> {
                new Task() {
                    TaskName = "Complete Project A",
                    StartDate = new DateTime(2009, 7, 17),
                    EndDate = new DateTime(2009, 7, 10)
                },
                new Task() {
                    TaskName = "Test Website",
                    StartDate = new DateTime(2009, 7, 10),
                    EndDate = new DateTime(2009, 7, 12)
                },
                new Task() {
                    TaskName = string.Empty,
                    StartDate = new DateTime(2009, 7, 4),
                    EndDate = new DateTime(2009, 7, 6)
                }
            };
        }

        [Command]
        public void ValidateRow(RowValidationArgs args) {
            args.Result = GetValidationErrorInfo((Task)args.Item);
        }
        static ValidationErrorInfo GetValidationErrorInfo(Task task) {
            if(task.StartDate > task.EndDate)
                return new ValidationErrorInfo("Start Date must be less than End Date");
            if(string.IsNullOrEmpty(task.TaskName))
                return new ValidationErrorInfo("Enter a task name");
            return null;
        }

        [Command]
        public void InvalidRow(InvalidRowExceptionArgs args) {
            args.ExceptionMode = ExceptionMode.NoAction;
        }
    }
}

File: VB/ValidateRow_MVVM/MainViewModel.vb
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.DataAnnotations
Imports DevExpress.Mvvm.Xpf
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel

Namespace ValidateRow_MVVM

    Public Class Task
        Implements IDataErrorInfo

        Public Property TaskName As String

        Public Property StartDate As Date

        Public Property EndDate As Date

        Private ReadOnly Property [Error] As String Implements IDataErrorInfo.[Error]
            Get
                Return If(StartDate > EndDate, "Either Start Date or End Date should be corrected.", Nothing)
            End Get
        End Property

        Private ReadOnly Property Item(ByVal columnName As String) As String Implements IDataErrorInfo.Item
            Get
                Select Case columnName
                    Case NameOf(Task.StartDate)
                        If StartDate > EndDate Then Return "Start Date must be less than End Date"
                    Case NameOf(Task.EndDate)
                        If StartDate > EndDate Then Return "End Date must be greater than Start Date"
                    Case NameOf(Task.TaskName)
                        If String.IsNullOrEmpty(TaskName) Then Return "Enter a task name"
                End Select

                Return Nothing
            End Get
        End Property
    End Class

    Public Class MainViewModel
        Inherits ViewModelBase

        Public ReadOnly Property TaskList As List(Of Task)

        Public Sub New()
            TaskList = New List(Of Task) From {New Task() With {.TaskName = "Complete Project A", .StartDate = New DateTime(2009, 7, 17), .EndDate = New DateTime(2009, 7, 10)}, New Task() With {.TaskName = "Test Website", .StartDate = New DateTime(2009, 7, 10), .EndDate = New DateTime(2009, 7, 12)}, New Task() With {.TaskName = String.Empty, .StartDate = New DateTime(2009, 7, 4), .EndDate = New DateTime(2009, 7, 6)}}
        End Sub

        <Command>
        Public Sub ValidateRow(ByVal args As RowValidationArgs)
            args.Result = MainViewModel.GetValidationErrorInfo(CType(args.Item, Task))
        End Sub

        Private Shared Function GetValidationErrorInfo(ByVal task As Task) As ValidationErrorInfo
            If task.StartDate > task.EndDate Then Return New ValidationErrorInfo("Start Date must be less than End Date")
            If String.IsNullOrEmpty(task.TaskName) Then Return New ValidationErrorInfo("Enter a task name")
            Return Nothing
        End Function

        <Command>
        Public Sub InvalidRow(ByVal args As InvalidRowExceptionArgs)
            args.ExceptionMode = ExceptionMode.NoAction
        End Sub
    End Class
End Namespace
